server/src/main/java/io/druid/server/SegmentManager.java

```diff
@@ -54,18 +95,22 @@ public SegmentManager(
     this.segmentLoader = segmentLoader;
   }
 
+  @VisibleForTesting
+  Map<String, DataSourceState> getDataSources()
+  {
+    return dataSources;
+  }
+
   public Map<String, Long> getDataSourceSizes()
```

Comment:
It sounds good, but it breaks the mocking behavior in ```HistoricalMetricsMonitorTest```. I don't see a good idea for this without refactoring it to not use mocking. 
I think copying is fine because each historical/realtime will have a small number of data sources. What do you think?